### PR TITLE
Refactor core-load-paths to use an in-line function

### PR DESCRIPTION
**Pull request by bryce-carson**
- Remove the two functions that add directories to the load path.
- Add a lambda funcall to the dolist and remove the call to the first of the two functions that this commit would remove.

### CODE REVIEW

**Comment by lebensterben** [comment]
- indentation isn't correct.
- no need to (funcall (lambda () ...). just modifying the load path suffices.

**Comment by bryce-carson** [comment]
@lebensterben is the PR ready now with 91a8938?

It wasn't *necessary*, but I thought making the variable names more explicit would be good to do as well.

**Comment by bryce-carson** [comment]
Wait... the `let` form is entirely unnecessary. One last commit then I'll leave it up to you.

File: core/core-load-paths.el
Comment (dalanicolai) [inline core/core-load-paths.el:121]: I guess, as previously, the 'error' alternative wasn't there, it is better to use `display-warning`  (or the shorter `lwarn`) here

Comment (lebensterben) [reply core/core-load-paths.el:121]: It's okay to use `error` here, because if any of the load paths does not exist, something is really wrong.

Comment (lebensterben) [reply core/core-load-paths.el:121]: It's okay to use `error` here, because if any of the load paths does not exist, something is really wrong.

**Review by smile13241324 — APPROVED** [review]

**Comment by smile13241324** [comment]
Looks good, still I want to do a quick test when I am back at home before merging.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request closed without merging**
